resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb

```diff
@@ -35,7 +35,7 @@ def detect # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength,
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::K8S_POD_NAME] = ENV['HOSTNAME'] || safe_gethostname
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::K8S_NODE_NAME] = gcp_env.lookup_metadata('instance', 'hostname')
 
-            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME']
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME'] if ENV['CONTAINER_NAME']
```

Comment:
Any reason not to do `# rubocop:disable Style/FetchEnvVar` here, as you did in the `cloud_platform_test`?

Reply:
very sorry those changes snuck in there. I have a separate PR for each linter rule that I am loading up